### ISSUE TITLE
Cannot run container service

### ISSUE THREAD

**Issue by kant2002** [microsoft/azure-pipelines-agent#2613]
## Agent Version and Platform
Version of your agent? @9187f14d2351fb7693ba6718824575c700dc7315

Window 1909 Client

## Azure DevOps Type and Version
dev.azure.com (formerly visualstudio.com)

## What's not working?
I try to define `container` based `service` in YAML file. I have to run version from master, to be able to run from Windows Client 1909 and not on server. I create and was able to initially start container.

```
Starting: Initialize containers
##[debug]Current Windows version: '1909 (Client)'
C:\Program Files\Docker\Docker\Resources\bin\docker.EXE version --format '{{.Server.APIVersion}}'
'1.40'
Docker daemon API version: '1.40'
C:\Program Files\Docker\Docker\Resources\bin\docker.EXE version --format '{{.Client.APIVersion}}'
'1.40'
Docker client API version: '1.40'
##[debug]Delete stale containers from previous jobs
C:\Program Files\Docker\Docker\Resources\bin\docker.EXE ps --all --quiet --no-trunc --filter "label=c18c3d"
##[debug]Delete stale container networks from previous jobs
C:\Program Files\Docker\Docker\Resources\bin\docker.EXE network prune --force --filter "label=c18c3d"
C:\Program Files\Docker\Docker\Resources\bin\docker.EXE pull mcr.microsoft.com/mssql/server:2019-GA-ubuntu-16.04
2019-GA-ubuntu-16.04: Pulling from mssql/server
Digest: sha256:c8fa22553ce421b0482febcafa712b29cbb933b0d97a8671686797b31cf157a9
Status: Image is up to date for mcr.microsoft.com/mssql/server:2019-GA-ubuntu-16.04
mcr.microsoft.com/mssql/server:2019-GA-ubuntu-16.04
C:\Program Files\Docker\Docker\Resources\bin\docker.EXE pull mcr.microsoft.com/mssql/server:2019-GA-ubuntu-16.04
2019-GA-ubuntu-16.04: Pulling from mssql/server
Digest: sha256:c8fa22553ce421b0482febcafa712b29cbb933b0d97a8671686797b31cf157a9
Status: Image is up to date for mcr.microsoft.com/mssql/server:2019-GA-ubuntu-16.04
mcr.microsoft.com/mssql/server:2019-GA-ubuntu-16.04
C:\Program Files\Docker\Docker\Resources\bin\docker.EXE inspect --format="{{.Os}}" mcr.microsoft.com/mssql/server:2019-GA-ubuntu-16.04
linux
##[error]Object reference not set to an instance of an object.
##[debug]System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.VisualStudio.Services.Agent.Worker.Container.DockerCommandManager.DockerNetworkCreate(IExecutionContext context, String network) in <agent_src>\src\Agent.Worker\Container\DockerCommandManager.cs:line 194
   at Microsoft.VisualStudio.Services.Agent.Worker.ContainerOperationProvider.CreateContainerNetworkAsync(IExecutionContext executionContext, String network) in <agent_src>\src\Agent.Worker\ContainerOperationProvider.cs:line 648
   at Microsoft.VisualStudio.Services.Agent.Worker.ContainerOperationProvider.StartContainersAsync(IExecutionContext executionContext, Object data) in <agent_src>\src\Agent.Worker\ContainerOperationProvider.cs:line 104
   at Microsoft.VisualStudio.Services.Agent.Worker.JobExtensionRunner.RunAsync() in <agent_src>\src\Agent.Worker\JobExtensionRunner.cs:line 36
   at Microsoft.VisualStudio.Services.Agent.Worker.StepsRunner.RunStepAsync(IStep step, CancellationToken jobCancellationToken) in <agent_src>\src\Agent.Worker\StepsRunner.cs:line 212
```

I define approximately following YAML. Everything run inside directory on host, not inside container.

```
resources:
  containers:
  - container: sqlserver
    image: mcr.microsoft.com/mssql/server:2019-GA-ubuntu-16.04
    env:
      ACCEPT_EULA: Y
      SA_PASSWORD: "YourStrong@Passw0rd>"
    ports:
      - 1433

jobs:
  - job:
     services:
        sqlserver: sqlserver
```

**Comment by kant2002** [comment]
cc @jtpetty

**Comment by kant2002** [comment]
It was reintroduced in 2.163.1

**Comment by jtpetty** [comment]
@kant2002 - We had to rerelease 2.160.1 as 2.163.1 to work around a different bug. We are working on releasing 2.164.0 that will have all the fixes since 2.160.1.

**Comment by kant2002** [comment]
Thanks. make sense now. Will revert my agent to latest version from master then.
